File: Assets/Scripts/GUI&HUD/CameraController.js
#pragma strict

var speed : int;
var maxX : int;
var maxZ : int;
var minS : int;
var maxS : int;

function Start () {

}

function Update () {
  var dx : float = Input.GetAxis("Horizontal") * Time.deltaTime * speed;
  var dz : float = Input.GetAxis("Vertical") * Time.deltaTime * speed;
  transform.position.x += dx;
  transform.position.z += dz;
  // scroll
  if (Input.GetAxis("Mouse ScrollWheel")) {
    Camera.main.orthographicSize -= Input.GetAxis("Mouse ScrollWheel") * Time.deltaTime * 4000;
  }
  // Camera boundary
  if (Camera.main.orthographicSize < minS) Camera.main.orthographicSize = minS;
  if (Camera.main.orthographicSize > maxS) Camera.main.orthographicSize = maxS;
  // Under construction
  var cameraLength : int = 0;
  if (transform.position.x < -maxX + cameraLength) transform.position.x = -maxX + cameraLength;
  if (transform.position.x > maxX - cameraLength) transform.position.x = maxX - cameraLength;
  if (transform.position.z < -maxZ + cameraLength) transform.position.z = -maxZ + cameraLength;
  if (transform.position.z > maxZ - cameraLength) transform.position.z = maxZ - cameraLength;
}

File: Assets/Scripts/new/Camera/CameraController.cs
using UnityEngine;
using System.Collections;

public class CameraController : MonoBehaviour {

  public float speed;

  public float scrollingSpeed;
  public float maxCameraSize;
  public float minCameraSize;

  public int maxEdgeDistanceToMove = 5;
  public float edgeMovingModifier = 0.5f;

  public Vector3 minBoundPostion;
  public Vector3 maxBoundPostion;

  void Update () {
    
    float deltaX = Input.GetAxis("Horizontal") * Time.deltaTime * speed;
    float deltaZ = Input.GetAxis("Vertical") * Time.deltaTime * speed;
    
    // Moving when mouse is close to the edge of the screen
    
    if (Input.mousePosition.x <= maxEdgeDistanceToMove) {
      deltaX = -edgeMovingModifier * speed * Time.deltaTime;
    }

    if (Input.mousePosition.y <= maxEdgeDistanceToMove) {
      deltaZ = -edgeMovingModifier * speed * Time.deltaTime;
    }

    if (Input.mousePosition.x >= Screen.width - maxEdgeDistanceToMove) {
      deltaX = edgeMovingModifier * speed * Time.deltaTime;
    }

    if (Input.mousePosition.y >= Screen.height - maxEdgeDistanceToMove) {
      deltaZ = edgeMovingModifier * speed * Time.deltaTime;
    }

    Vector3 newPosition = transform.position + new Vector3(deltaX, 0, deltaZ);

    newPosition = Vector3.Max(newPosition, minBoundPostion);
    newPosition = Vector3.Min(newPosition, maxBoundPostion);

    transform.position = newPosition;

    // Scroll

    if (Input.GetAxis("Mouse ScrollWheel") != 0) {
      Camera.main.orthographicSize -= Input.GetAxis("Mouse ScrollWheel") * Time.deltaTime * scrollingSpeed;
    }

    Camera.main.orthographicSize = Mathf.Max(Camera.main.orthographicSize, minCameraSize);
    Camera.main.orthographicSize = Mathf.Min(Camera.main.orthographicSize, maxCameraSize);

  }

}
